src/shared/userdb.h

```diff
@@ -39,19 +40,19 @@ typedef enum UserDBFlags {
  *  -ETIMEDOUT: Time-out
  */
 
-int userdb_by_name(const char *name, UserDBFlags flags, UserRecord **ret);
-int userdb_by_uid(uid_t uid, UserDBFlags flags, UserRecord **ret);
-int userdb_all(UserDBFlags flags, UserDBIterator **ret);
+int userdb_by_name(Varlink *v, const char *name, UserDBFlags flags, UserRecord **ret);
+int userdb_by_uid(Varlink *v, uid_t uid, UserDBFlags flags, UserRecord **ret);
+int userdb_all(Varlink *v, UserDBFlags flags, UserDBIterator **ret);
```

Comment:
hmm, instead of adding this extra param to all functions and then make most callers pass NULL to it, maybe instead rename the functions to `userdb_all_with_impersonation()` which takes the extra arg, and then add a static inline wrapper with the old name, that passes NULL on that new arg:

```c
int userdb_all_with_impersonation(Varlink *v, UserDBFlags flags, UserDBIterator **ret);

static inline userdb_all(UserDBFlags flags, UserDBIterator **ret) {
        return userdb_all_with_impersonation(NULL, flags, ret);
}
```

you get the idea. that hides the new arg inside of userdb.[ch] for most cases, and keeps regular callers simple.